routers/routes/web.go

```diff
@@ -162,6 +162,9 @@ func WebRoutes() *web.Route {
 	// We use r.Route here over r.Use because this prevents requests that are not for avatars having to go through this additional handler
 	routes.Route("/avatars/*", "GET, HEAD", storageHandler(setting.Avatar.Storage, "avatars", storage.Avatars))
 	routes.Route("/repo-avatars/*", "GET, HEAD", storageHandler(setting.RepoAvatar.Storage, "repo-avatars", storage.RepoAvatars))
+	routes.Route("/serviceworker.js", "GET, HEAD", http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		http.Redirect(resp, req, setting.AppURL+"assets/serviceworker.js", http.StatusFound)
```

Comment:
```suggestion
		http.Redirect(resp, req, setting.AppURL+"assets/serviceworker.js", http.StatusTemporaryRedirect)
```

We should only use 307 or 308 redirects because those forbid method change which is better for security.

Reply:
I'm not certain I understand how security or functionality could affected by posts etc being redirected to assets - in fact it would be better if they were.

Reply:
Does the web browser support the redirect?

Reply:
It does not matter for GET requests but for other methods, browsers may decide to change method to GET based on obscure criteria related to HTML forms which I think is an outdated concept. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections#temporary_redirections for more details.

Security can be impacted of POST to a endpoint with credentials where a change to GET would move potentially sensitive parameters from request body to request URL, adding them to the browser history and making them visible in server logs. For that reason alone, I never use 301, 302 or 303, only 307 and 308 which of course all browsers support.